luaui/Widgets/api_analytics.lua

```diff
@@ -0,0 +1,96 @@
+local widget = widget --- @type Widget
+
+function widget:GetInfo()
+	return {
+		name    = "Analytics API",
+		desc    = "Provides an API for sending analytics events using SendLuaUIMsg ",
+		author  = "uBdead",
+		date    = "Oct 2025",
+		license = "GPL-v2",
+		layer   = -1,
+		enabled = true,
+	}
+end
+
+local spGetPlayerInfo = Spring.GetPlayerInfo
+local spGetMyPlayerID = Spring.GetMyPlayerID
+local spGetGameFrame = Spring.GetGameFrame
+local spLog = Spring.Log
+local spSendLuaUIMsg = Spring.SendLuaUIMsg
+
+local pendingAnalytics = {}
+
+local function analyticsCoroutine(eventType, eventData)
+	local myPlayerName = spGetPlayerInfo(spGetMyPlayerID())
+	local frameNumber = spGetGameFrame()
+	local jsondict = {}
+	-- Add static fields
+	jsondict.eventtype = eventType
+	jsondict.username = myPlayerName
+	jsondict.frame = frameNumber
+
+	-- Add eventData incrementally
+	if type(eventData) == "table" then
+		local keys = {}
+		for k in pairs(eventData) do keys[#keys+1] = k end
+		local i = 1
+		while i <= #keys do
+			local k = keys[i]
+			if jsondict[k] == nil then
+				jsondict[k] = eventData[k]
+			else
+				spLog("Analytics API: Key conflict for '" .. k .. "'. Skipping this key.", Spring.LOG_WARNING)
+			end
+			i = i + 1
+		end
+	end
+
+	local jsonstr = Json.encode(jsondict)
+	coroutine.yield()
+	
+	local b64str = string.base64Encode(jsonstr)
+	coroutine.yield()
+
+	local complexMatchEvent = "complex-match-event:" .. b64str
+	-- Spring.Echo(complexMatchEvent)
+	spSendLuaUIMsg(complexMatchEvent)
+end
+
+local function sendAnalyticsEvent(eventType, eventData)
+	local co = coroutine.create(function() analyticsCoroutine(eventType, eventData) end)
+	table.insert(pendingAnalytics, co)
+end
+
+function widget:Update()
+	-- Process one step of each pending analytics coroutine per frame
+	local finished = {}
+	for i, co in ipairs(pendingAnalytics) do
+		local status, res = coroutine.resume(co)
+		if coroutine.status(co) == "dead" then
+			finished[#finished+1] = i
+		end
+	end
+	-- Remove finished coroutines
+	for j = #finished, 1, -1 do
+		table.remove(pendingAnalytics, finished[j])
+	end
+end
+
```

Comment:
consider cleaning up
```suggestion
function widget:Shutdown()
	WG.Analytics = nil
end
```